bcipy/io/load.py

```diff
@@ -318,3 +302,297 @@ def fast_scandir(directory_name: str, return_path: bool = True) -> List[str]:
+        self.info = {
+            'user_id': user_id,
+            'date': date,
+            'experiment_id': self.experiment_id,
+            'date_time': self.date_time,
+            'task': task,
+            'task_name': self.task_name,
+            'run': run,
+            'path': path
+        }
+
+    def get_parameters(self) -> Parameters:
+        return load_json_parameters(
+            f'{self.path}/{DEFAULT_PARAMETERS_FILENAME}',
+            value_cast=True)
+
+    def __str__(self):
+        return f'BciPySessionTaskData: {self.info=}'
+
+    def __repr__(self):
+        return f'BciPySessionTaskData: {self.info=}'
+
+
+class BciPyCollection:
+    """BciPy Data.
+
+    This class is used to represent a full BciPy data collection.
+    """
+
+    def __init__(
+            self,
+            data_directory: str,
+            experiment_id: Optional[str] = None,
+            user_id: Optional[str] = None,
+            date: Optional[str] = None,
+            date_time: Optional[str] = None,
+            excluded_tasks: Optional[List[str]] = None,
+            anonymize: bool = False) -> None:
+        if not os.path.isdir(data_directory):
+            raise FileNotFoundError(
+                f'Data directory not found at [{data_directory}]')
+        self.data_directory = data_directory
+
+        self.experiment_id = experiment_id
+        self.user_id = user_id
+        self.date = date
+        self.date_time = date_time
+        self.excluded_tasks = excluded_tasks
+        self.anonymize = anonymize
+
+        self.session_task_data = []
+        self.user_paths = []
+        self.users = []
+
+        self.date_paths = []
+        self.dates = []
+
+        self.experiment_paths = []
+        self.experiments = []
+
+        self.date_time_paths = []
+        self.date_times = []
+
+        self.task_paths = []
+        self.tasks = []
+
+    def collect(self) -> List[BciPySessionTaskData]:
+        if not self.session_task_data:
+            self.load_tasks()
+
+        # if anonymize is set, return the anonymized data. Make a map of user_id to anonymized id
+        if self.anonymize:
+            user_map = {}
+            user_id_increment = 1
+            for task in self.session_task_data:
+                if task.user_id not in user_map:
+                    user_map[task.user_id] = f'ID{user_id_increment}'
+                    user_id_increment += 1
+                task.user_id = user_map[task.user_id]
+
+        return self.session_task_data
+
+    def load_users(self):
+        user_paths = fast_scandir(self.data_directory, return_path=True)
+        users = []
+        for user in user_paths:
+            if self.user_id:
+                if self.user_id in user:
+                    users.append(user)
+            else:
+                users.append(user)
+
+        self.user_paths = users
+        self.users = [user.split('/')[-1] for user in users]
```

Comment:
These `if` statements could be collapsed to `if self.user_id and self.user_id in user`. However, I think the whole `for` loop can be skipped if there is no `self.user_id` filter defined.

```
if self.user_id:
    self.user_paths = [user for user in user_paths if self.user_id in user]
else:
    self.user_paths = user_paths
self.users = [user.split('/')[-1] for user in self.user_paths
```
Or alternatively self.users could be a computed property. The same pattern applies to most of the filtering methods in the class.